packages/block/src/block.ts

```diff
@@ -223,44 +224,57 @@ export class Block {
   }
 
   /**
-   *  Method to retrieve a block from the provider and format as a {@link Block}
-   * @param provider an Ethers JsonRPCProvider
+   *  Method to retrieve a block from a JSON-RPC provider and format as a {@link Block}
+   * @param provider either a url for a remote provider or an Ethers JsonRpcProvider object
    * @param blockTag block hash or block number to be run
    * @param opts {@link BlockOptions}
    * @returns the block specified by `blockTag`
    */
-  public static fromEthersProvider = async (
-    provider: ethers.providers.JsonRpcProvider | string,
+  public static fromJsonRpcProvider = async (
+    provider: any,
     blockTag: string | bigint,
     opts: BlockOptions
   ) => {
     let blockData
-    const prov =
-      typeof provider === 'string' ? new ethers.providers.JsonRpcProvider(provider) : provider
+    const providerUrl = getProvider(provider)
+
     if (typeof blockTag === 'string' && blockTag.length === 66) {
-      blockData = await prov.send('eth_getBlockByHash', [blockTag, true])
+      blockData = await fetchFromProvider(providerUrl, {
+        method: 'eth_getBlockByHash',
+        params: [blockTag, true],
+      })
     } else if (typeof blockTag === 'bigint') {
-      blockData = await prov.send('eth_getBlockByNumber', [bigIntToHex(blockTag), true])
+      blockData = await fetchFromProvider(providerUrl, {
+        method: 'eth_getBlockByNumber',
+        params: [bigIntToHex(blockTag), true],
+      })
     } else if (
       isHexPrefixed(blockTag) ||
       blockTag === 'latest' ||
       blockTag === 'earliest' ||
       blockTag === 'pending'
```

Comment:
Beyond the scope of this PR but maybe we should check if `blockTag` is of any kind of whatever valid string values there are? I think `syncing` is a new param you can use as blockTag as of, I think, the merge?

Reply:
ummm what would `syncing` point to? since merge we can now use `finalized` and `safe`, we can add them
```suggestion
      blockTag === 'pending' ||
      blockTag === 'finalized' ||
      blockTag === 'safe'
```

Reply:
Sorry, I meant `safe`.